Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -114,8 +115,33 @@ export class ArtifactEngine {
         }
     }
 
+    shouldDownload(path: string) {
+        var shouldDownload = false;
+        for (let pattern of this.patternList) {
+            if (pattern.charAt(0) === '-' && (minimatch(path, pattern.substr(1), { dot: true, nocase: true }))) {
+                return false;
+            }
+            else {
+                if (minimatch(path, (pattern.charAt(0) === '+') ? pattern.substr(1) : pattern, { dot: true, nocase: true })) {
+                    shouldDownload = true;
+                }
+            }
+        }
+        return shouldDownload;
+    }
+
+    createPatternList(artifactEngineOptions: ArtifactEngineOptions) {
```

Comment:
good to add comment on what sort of itemPatterns are supported with examples
i suppose we can give the following examples:

a. none (in which case we assume **)
b. **
c. drop\starstar
d. "+drop\starstar","-drop\star.exe"
e. "+drop\foo\starstar","-drop\foo\bar\starstar"

PS: replace star with * (i had trouble adding ** as it was not showing up properly in comment)